pkg/cmd/roachtest/tests/logical_data_replication.go

```diff
@@ -230,6 +244,66 @@ func TestLDROnNodeShutdown(
 	VerifyCorrectness(t, setup, leftJobID, rightJobID, 5*time.Minute)
 }
 
+// TestLDROnNetworkPartition aims to see what happens when both clusters
+// are separated from one another by a network partition. This test will
+// aim to keep the workload going on both sides and wait for reconciliation
+// once the network partition has completed
+func TestLDROnNetworkPartition(
+	ctx context.Context, t test.Test, c cluster.Cluster, setup multiClusterSetup,
+) {
+
+	duration := 10 * time.Minute
+	if c.IsLocal() {
+		duration = 3 * time.Minute
+	}
+
+	kvWorkload := replicateKV{
+		readPercent:             0,
+		debugRunDuration:        duration,
+		maxBlockBytes:           1024,
+		initRows:                1000,
+		initWithSplitAndScatter: true}
+
+	leftJobID, rightJobID := setupLDR(ctx, t, c, setup, kvWorkload)
+
+	monitor := c.NewMonitor(ctx, setup.CRDBNodes())
+	monitor.Go(func(ctx context.Context) error {
+		return c.RunE(ctx, option.WithNodes(setup.workloadNode), kvWorkload.sourceRunCmd("system", setup.CRDBNodes()))
+	})
+
+	// Let workload run for a bit before we kill a node
+	time.Sleep(kvWorkload.debugRunDuration / 10)
+
+	// Partition the left and right clusters so they can still make progress relative to each other,
+	// but can't talk to the nodes on the other side of the partition.
+	// Normally, the blackholeFailer is accessed through the failer interface,
+	// at least in the failover tests. Because this test shouldn't use all the
+	// failer interface calls (e.g. Setup(), and Ready()), we use the
+	// blakholeFailer struct directly. In other words, in this test, we
+	// shouldn't treat the blackholeFailer as an abstracted api.
+
+	disconnectDuration := kvWorkload.debugRunDuration / 5
+	t.L().Printf("Disconnecting left and right nodes for %.2f minutes", disconnectDuration.Minutes())
+	blackholeFailer := &blackholeFailer{t: t, c: c, input: true, output: true}
+	for _, nodeID := range setup.left.nodes {
+		blackholeFailer.FailPartial(ctx, nodeID, setup.right.nodes)
+	}
+
+	for _, nodeID := range setup.right.nodes {
+		blackholeFailer.FailPartial(ctx, nodeID, setup.left.nodes)
+	}
```

Comment:
Hrm. I wonder if the more interesting test here is a partial network failure rather than disconnecting the full set. Also, I don't think we should have to do this in both directions since we've set input and output to true?

Reply:
Ack on the both directions part, I'll fix that.

I was aiming for a datacenter network failure type scenario. But I can make it random in how many nodes are isolated and it should cover all scenarios